### PR TITLE
Refactor the Application configuration startup code

### PR DESCRIPTION
**Pull request by mregen**
## Purpose

* Refactor the Application configuration startup code, to address issues with cert creation and tbd enhancements of the UA stack.
* Latest UA stack fixes a certificate creation issue in windows containers

## Does this introduce a breaking change?
```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code, start the app

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

## What to Check
Verify that the following are valid
* opc plc startup code works as usual

### CODE REVIEW

File: src/OpcApplicationConfiguration.cs [hunk 1]
```diff
@@ -75,117 +77,117 @@ public OpcApplicationConfiguration()
...
+                Logger.Information($"Security policy {policy.SecurityPolicyUri} with mode {policy.SecurityMode} added");
+                if (policy.SecurityMode == MessageSecurityMode.None)
+                {
+                    Logger.Warning("Note: This is a security risk and needs to be disabled for production use");
+                }
+            }
 
-            // show certificate store information
-            await ShowCertificateStoreInformationAsync().ConfigureAwait(false);
+            Logger.Information($"LDS(-ME) registration intervall set to {LdsRegistrationInterval} ms (0 means no registration)");
```
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:130]: typo interval

File: src/OpcApplicationConfiguration.cs [hunk 2]
```diff
@@ -75,117 +77,117 @@ public OpcApplicationConfiguration()
         /// </summary>
         public async Task<ApplicationConfiguration> ConfigureAsync()
         {
-            // instead of using a configuration XML file, we configure everything programmatically
-
-            // passed in as command line argument
-            ApplicationConfiguration = new ApplicationConfiguration
-            {
+            // instead of using a configuration XML file, configure everything programmatically
+            var application = new ApplicationInstance() {
```
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:81]: We use curly braces in the next line for this solution. Please reformat document.

File: src/OpcApplicationConfiguration.cs [hunk 3]
```diff
@@ -75,117 +77,117 @@ public OpcApplicationConfiguration()
...
+            // instead of using a configuration XML file, configure everything programmatically
+            var application = new ApplicationInstance() {
                 ApplicationName = ApplicationName,
-                ApplicationUri = ApplicationUri,
-                ProductUri = ProductUri,
-                ApplicationType = ApplicationType.Server,
-
-                // configure OPC stack tracing
-                TraceConfiguration = new TraceConfiguration()
+                ApplicationType = ApplicationType.Server
```
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:83]: Add trailing comma

File: src/OpcApplicationConfiguration.cs (hunk 2 above)
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:81]: Cosmetic: Remove ()

File: src/OpcApplicationConfiguration.cs [hunk 4]
```diff
@@ -75,117 +77,117 @@ public OpcApplicationConfiguration()
...
-            // Support larger number of nodes.
-            ApplicationConfiguration.ServerConfiguration.MaxMessageQueueSize = MAX_MESSAGE_QUEUE_SIZE;
-            ApplicationConfiguration.ServerConfiguration.MaxNotificationsPerPublish = MAX_NOTIFICATIONS_PER_PUBLISH;
-            ApplicationConfiguration.ServerConfiguration.MaxSubscriptionCount = MAX_SUBSCRIPTION_COUNT;
-            ApplicationConfiguration.ServerConfiguration.MaxPublishRequestCount = MAX_PUBLISH_REQUEST_COUNT;
-            ApplicationConfiguration.ServerConfiguration.MaxRequestThreadCount = MAX_REQUEST_THREAD_COUNT;
+            // configure OPC stack tracing
+            Utils.SetTraceMask(OpcStackTraceMask);
+            Utils.Tracing.TraceEventHandler += LoggerOpcUaTraceHandler;
+            Logger.Information($"opcstacktracemask set to: 0x{OpcStackTraceMask:X}");
```
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:135]: Is there a more meaningful name for opcstacktracemask that we can log here?

File: src/OpcApplicationConfiguration.cs [hunk 5]
```diff
@@ -75,117 +77,117 @@ public OpcApplicationConfiguration()
...
 
-            return ApplicationConfiguration;
-        }
+            // log certificate status
+            var certificate = ApplicationConfiguration.SecurityConfiguration.ApplicationCertificate.Certificate;
+            if (certificate == null)
+            {
+                Logger.Information($"No existing Application certificate found. Create a self-signed Application certificate valid from yesterday for {CertificateFactory.DefaultLifeTime} months,");
+                Logger.Information($"with a {CertificateFactory.DefaultKeySize} bit key and {CertificateFactory.DefaultHashSize} bit hash.");
```
Comment (luiscantero) [inline src/OpcApplicationConfiguration.cs:142]: This will generate two log lines with two different timestamps. Pls merge the string instead.

**Review by luiscantero — CHANGES_REQUESTED** [review]

File: src/Program.cs [hunk 6]
```diff
@@ -14,6 +14,7 @@
 using System.Linq;
 using System.Net;
 using System.Reflection;
+using System.Text;
 using System.Threading;
```
Comment (mregen) [inline src/Program.cs:18]: wohoo

**Comment by mregen** [comment]
@luiscantero all resolved

**Review by luiscantero — APPROVED** [review]

**Pull request merged**